validator/iam_custom_role_permissions.rego

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPIAMCustomRolePermissionsConstraintV1
+
+import data.validator.gcp.lib as lib
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	lib.get_constraint_params(constraint, params)
+	asset := input.asset
+
+	asset.asset_type == "iam.googleapis.com/Role"
+
+	asset_permissions := asset.resource.data.includedPermissions[_]
+	asset_title := asset.resource.data.title
+	lower(asset_title) == lower(params.title)
```

Comment:
Should we make this check optional? It seems like having to choose a specific role title might be messy. In many scenarios we might want to block a permission from all roles.

Maybe make this a regex w/ a default of matching everything?

Reply:
That's a good point. I created a default of "*" if title is not defined, so it will just look at all titles.

Reply:
Appreciate the quick review!